### DISCUSSION TITLE
Roadmap 2023

### DISCUSSION THREAD

**Discussion by gwbres** [nav-solutions/rinex#97] [General]
Roadmap : tasks we can reasonnably accomplish by the end of this year

- [ ] Preprocessing
  - [ ] Fix the Hatch filter operation. 
It is a very convenient algorithm to smooth Pseudo Range observations. 
This will probably prove very important if we move towards a PVT solver for example.
  - [ ] Provide Phase Origin alignment as a possible preprocessing operation
  - [ ] Provide Phase Offset as a new possible preprocessing operation
- [ ] Investigate Cycle Slip detection algorithms
- [ ] Parser :
  - [ ] #81 
  - [ ] #76 
  - [ ] #80 
- [ ] #69 
- [ ] Writer / generator
  - [ ] Improve the Data generation interface in the command line interface
- [ ] Documentation
  - [ ] Document IONEX usage in the command line
- [ ] Conclude the python bindings ?
Currently the python bindings PR is solely stuck due to complex Enums that cannot be converted to PyObjects directly.
A work around is to write Python compatible versions of these problematic structures and provide .to() and .from().
This is initiated, I'll see if I can conclude this topic

**Comment by larsnaesbye** [comment]
A good use case for `rinex-cli` is to be a modern replacement for `teqc`. A guide describing how to mimic tecq behaviour would be incredibly useful for many people.

**Reply by larsnaesbye** [reply]
Hi @gwbres , 

I'm thinking about writing that guide to assist people in moving from `teqc`. Right now, just a structure document [here](https://github.com/larsnaesbye/rinex/blob/main/doc/rinex-cli%20for%20teqcies.md).

Any thoughts?

**Reply by gwbres** [reply]
Hello,

yes, basically you can extract things from the existing processing/gnss/qc part of the documentation and provide meaningful examples.
Currently we host CRNX/V3/ESBCDNK and CRNX/V3/MOJDNK with associated NAV files that you provided, which is enough for meaningful examples, unfortunately not enough to resolve a position at the moment but that should change in the future

**Reply by larsnaesbye** [reply]
Good. I have begun a draft PR and will add to it slowly.
